### PR TITLE
Trainee booking with API

### PR DESCRIPTION
**Pull request by nnanthicha**
- add booking API
- remove "calendar" and "start chat" button if login as "Trainer"
- add booking complete page

### CODE REVIEW

File: src/services/booking.service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import authHeader from "./auth-header";
+
+const API_URL = process.env.NEXT_PUBLIC_API_URL
+    ? process.env.NEXT_PUBLIC_API_URL
+    : "";
+
+export type BookingInput = {
+    date: string,
+    endTime: string,
+    startTime: string,
```
Comment (noppakorn) [inline src/services/booking.service.ts:11]: Should this be a datetime

File: src/pages/user/booking/[username]/index.tsx [hunk 2]
```diff
@@ -12,6 +12,14 @@ import Image from "next/image";
 const BookTrainerProfile = () => {
   const router = useRouter();
   const { username } = router.query;
+  const [profile, setProfile] = useState<UserProfile>();
+  useEffect(() => {
+    setLoading(true);
+    getCurrentUserProfile().then((data) => {
+      setProfile(data);
+    });
```
Comment (noppakorn) [inline src/pages/user/booking/[username]/index.tsx:20]: Loading never get set back to false?

**Review by noppakorn — CHANGES_REQUESTED** [review]

File: src/services/booking.service.ts (hunk 1 above)
Comment (nnanthicha) [reply src/services/booking.service.ts:11]: but backend required data as string

Comment (noppakorn) [reply src/services/booking.service.ts:11]: Ok, my mistake.

**Review by noppakorn — APPROVED** [review]

**Pull request merged**